pkg/kubelet/dockertools/manager_test.go

```diff
@@ -87,3 +113,55 @@ func TestSetEntrypointAndCommand(t *testing.T) {
 		}
 	}
 }
+
+func TestConvertDockerToRuntimeContainer(t *testing.T) {
+	dockerContainer := &docker.APIContainers{
+		ID:      "ab2cdf",
+		Image:   "bar_image",
+		Created: 12345,
+		Names:   []string{"/k8s_bar.5678_foo_ns_1234_42"},
+	}
+	expected := &kubecontainer.Container{
+		ID:      types.UID("ab2cdf"),
+		Name:    "bar",
+		Image:   "bar_image",
+		Hash:    0x5678,
+		Created: 12345,
+	}
+
+	actual, err := convertDockerToRuntimeContainer(dockerContainer)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+	if !reflect.DeepEqual(expected, actual) {
+		t.Errorf("expected %#v, got %#v", expected, actual)
+	}
+}
+
+func TestGetContainers(t *testing.T) {
+	manager, fakeDocker := NewFakeDockerManager()
+	containers := []docker.APIContainers{
+		{
+			ID:    "1234",
+			Names: []string{"/k8s_foo_qux_new_1234_42"},
+		},
+		{
+			ID:    "5678",
+			Names: []string{"/k8s_bar_qux_new_5678_42"},
+		},
+	}
+	expected := util.NewStringSet([]string{"foo", "bar"}...)
+	fakeDocker.ContainerList = containers
+	actualContainers, err := manager.GetContainers(false)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+
+	actual := util.NewStringSet()
+	for _, c := range actualContainers {
+		actual.Insert(c.Name)
+	}
+	if !reflect.DeepEqual(expected.List(), actual.List()) {
```

Comment:
`util.StringSet` actually sorts this https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/util/set.go#L122